crates/iota-core/src/execution_cache.rs

```diff
@@ -112,31 +112,12 @@ pub enum ExecutionCacheConfigType {
     PassthroughCache,
 }
 
-pub fn choose_execution_cache(config: &ExecutionCacheConfig) -> ExecutionCacheConfigType {
-    #[cfg(msim)]
-    {
-        let mut use_random_cache = None;
-        iota_macros::fail_point_if!("select-random-cache", || {
-            let random = rand::random::<bool>();
-            tracing::info!("Randomly selecting cache: {}", random);
-            use_random_cache = Some(random);
-        });
-        if let Some(random) = use_random_cache {
-            if random {
-                return ExecutionCacheConfigType::PassthroughCache;
-            } else {
-                return ExecutionCacheConfigType::WritebackCache;
-            }
-        }
+pub fn choose_execution_cache(_: &ExecutionCacheConfig) -> ExecutionCacheConfigType {
+    if std::env::var(DISABLE_WRITEBACK_CACHE_ENV_VAR).is_ok() {
+        error!("DISABLE_WRITEBACK_CACHE is no longer respected. WritebackCache is the default.");
     }
 
-    if std::env::var(DISABLE_WRITEBACK_CACHE_ENV_VAR).is_ok()
-        || matches!(config, ExecutionCacheConfig::PassthroughCache)
-    {
-        ExecutionCacheConfigType::PassthroughCache
-    } else {
-        ExecutionCacheConfigType::WritebackCache
-    }
+    ExecutionCacheConfigType::WritebackCache
```

Comment:
I would revert this change here, So we can still specify PassthroughCache in the config if we want.
(the use random cache for simtests can be removed)